x-pack/plugins/security_solution/public/timelines/components/timeline/event_details.tsx

```diff
@@ -11,39 +11,67 @@
  */
 
 import { EuiSpacer } from '@elastic/eui';
-import React from 'react';
+import React, { useMemo } from 'react';
 import deepEqual from 'fast-deep-equal';
+import { findIndex, pick } from 'lodash/fp';
 
 import { BrowserFields, DocValueFields } from '../../../common/containers/source';
 import {
   ExpandableEvent,
   ExpandableEventTitle,
+  HandleOnEventClosed,
 } from '../../../timelines/components/timeline/expandable_event';
 import { useDeepEqualSelector } from '../../../common/hooks/use_selector';
+import { useTimelineEventsDetails } from '../../containers/details';
+import { timelineSelectors } from '../../store/timeline';
+import { timelineDefaults } from '../../store/timeline/defaults';
 
 interface EventDetailsProps {
   browserFields: BrowserFields;
   docValueFields: DocValueFields[];
   timelineId: string;
+  handleOnEventClosed?: HandleOnEventClosed;
 }
 
 const EventDetailsComponent: React.FC<EventDetailsProps> = ({
   browserFields,
   docValueFields,
   timelineId,
+  handleOnEventClosed,
 }) => {
-  const expandedEvent = useDeepEqualSelector(
-    (state) => state.timeline.timelineById[timelineId]?.expandedEvent
+  const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
+  const { expandedEvent } = useDeepEqualSelector((state) =>
+    pick(['expandedEvent'], getTimeline(state, timelineId) ?? timelineDefaults)
```

Comment:
if we need only one prop it's better to use `useShallowEqualSelector and just access it directly, without using pick, like ```=> (getTimeline(state, timelineId) ?? timelineDefaults).expandedEvent```